### PR TITLE
Respect LTI language if present

### PR DESCRIPTION
**Pull request by niknetniko**
This pull request implements #2266.

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -227,4 +227,15 @@ def page_navigation_links(pages, remote = false, controller = '', params = {}, a
     end
   end
   # rubocop:enable Metrics/ParameterLists
+
+  def set_locale(locale, user)
```
Comment (thepieterdc) [inline app/helpers/application_helper.rb:231]: Is the `user` argument required here if this is only called for `current_user`?

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -122,12 +122,7 @@ def user_not_authorized
   end
 
   def set_locale
-    begin
-      I18n.locale = params[:locale] || current_user&.lang || I18n.default_locale
-    rescue I18n::InvalidLocale
-      I18n.locale = I18n.default_locale
-    end
-    current_user&.update(lang: I18n.locale.to_s) if current_user&.lang != I18n.locale.to_s
+    helpers.set_locale(params[:locale] || current_user&.lang || I18n.default_locale, current_user)
```
Comment (charvp) [inline app/controllers/application_controller.rb:125]: I don't think `helpers.` is required here?

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (niknetniko) [reply app/helpers/application_helper.rb:231]: This was done to be able to call the helper from the tests without setting up the whole authentication stuff. There is probably a way to do this, but I haven't found it yet.

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (niknetniko) [reply app/controllers/application_controller.rb:125]: This doesn't seem to work without, since the controller function is also called `set_locale`

Comment (charvp) [reply app/controllers/application_controller.rb:125]: Ah, good point.

**Review by bmesuere — APPROVED** [review]

**Pull request merged**